python/SHIELD_ADVANCED_ENABLED_AUTORENEW/SHIELD_ADVANCED_ENABLED_AUTORENEW.py

```diff
@@ -0,0 +1,389 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+'''
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  SHIELD_ADVANCED_ENABLED_AUTORENEW
+
+Description:
+  Check whether AWS Shield Advanced is enabled and is set to autorenew. As the API Endpoint of this service is only available in us-east-1, this rule should only be scheduled to run in us-east-1 region.
+
+Trigger:
+  Periodic
+
+Reports on:
+  AWS::::Account
+
+Scenarios:
+ Scenario: 1
+    Given: AWS Shield Advanced is not enabled
+    Then: Return NON_COMPLIANT with annotation "AWS Shield Advanced is not enabled."
+  Scenario: 2
+    Given: AWS Shield Advanced is enabled
+    And: Autorenew is DISABLED
+    Then: Return NON_COMPLIANT with annotation "AWS Shield Advanced subscription is not set to Autorenew."
+  Scenario: 3
+    Given: AWS Shield Advanced is enabled
+    And: Autorenew is ENABLED
+    Then: Return COMPLIANT
+'''
+import json
+import sys
+import datetime
+import boto3
+import botocore
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def shield_advanced_check(client):
+    try:
+        return client.describe_subscription()['Subscription']['AutoRenew']
+    except botocore.exceptions.ClientError as error:
+        return error.response['Error']['Code']
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    shield_advanced_enabled = shield_advanced_check(get_client('shield', event))
+    if shield_advanced_enabled == 'ResourceNotFoundException':
+        return [build_evaluation(event['accountId'],

```

Comment:
No need to return a list. Idem for the rest

Reply:
@jongogogo, it's done. Kindly review it